default/scripting/policies/INDOCTRINATION.focs.txt

```diff
@@ -9,19 +9,36 @@ Policy
     effectsgroups = [
         [[SPECIES_LIKES_OR_DISLIKES_POLICY_STABILITY_EFFECTS]]
 
-        // makes planets more stable over time at cost of influence
+        // makes non-influence-focused planets more stable over time at the cost of influence
         EffectsGroup
         scope = And [
             Planet
             OwnedBy empire = Source.Owner
             Species
+            Not Focus type = "FOCUS_INFLUENCE"
         ]
         effects = [
             SetTargetHappiness value = Value + 
                 min((NamedReal name = "PLC_INDOCTRINATION_STRENGTH_MAX" value = 10.0),
                     (CurrentTurn - TurnPolicyAdopted empire = Source.Owner name = "PLC_INDOCTRINATION") / 4.0)
+            SetTargetInfluence value = Value - 1.0
+        ]
 
-            SetTargetInfluence value = Value - 4.0
+        // makes influence-focused planets more stable over time and increases its influence
+        // TODO: Consider restricting this effect also to having a Reg. Adm. center.
+        EffectsGroup
+        scope = And [
+            Planet
+            OwnedBy empire = Source.Owner
+            Species
+            Focus type = "FOCUS_INFLUENCE"
+        ]
+        effects = [
+            SetTargetHappiness value = Value + 
+                min((NamedReal name = "PLC_INDOCTRINATION_STRENGTH_MAX" value = 10.0),
+                    (CurrentTurn - TurnPolicyAdopted empire = Source.Owner name = "PLC_INDOCTRINATION") / 4.0)
```

Comment:
Using `CurrentTurn - Something` or similar like this lead to off-by-one issues and lingering "Unknown" meter contributions for me. Does this use not have that issue?

Previous workaround was in https://github.com/freeorion/freeorion/commit/dcfc9642827948cc14c42df401ec6832c19fffee#diff-c9e61399cfb9d4f7c25c1a0a74e5171f2e9fce93ca1392fe7aef7a8e40d00b65

Reply:
I don't know, I haven't got to adopt this policy in my games yet.

Reply:
Please test.

A no-AI single player game with the cheap policies and cheap/fast techs test rules enabled would facilitate doing this quickly.

Reply:
I'll do and report back.

Reply:
When Indoctrination is adopted, before turning, Stability meters get a +0.0 from the policy.
Next turn it gets a +0.25 from Indoctrination and -0.25 from unknown.
Following turns it gets +0.5, +0.75, etc. while the unknown -0.25 stays fixed.

Do I have to wait for a workaround like TurnsSincePolicyAdopted empire = X name = Y?

Meanwhile I noticed that I didn't set a priority for the second effect, and I should, to avoid doubling influence upkeep when planet is set to influence. Or maybe move influence upkeep later in the effects execution.

So I'm setting this as work in progress.

Reply:
> Do I have to wait for a workaround like TurnsSincePolicyAdopted empire = X name = Y?

probably